### PR TITLE
[DOCS] Adds delete forecast API

### PR DESCRIPTION
**Pull request by lcawl**
Related to https://github.com/elastic/elasticsearch/pull/33218 and https://github.com/elastic/elasticsearch/issues/31134

### CODE REVIEW

File: docs/reference/ml/apis/delete-forecast.asciidoc [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+
+==== Path Parameters
+
+`job_id` (required)::
+  (string) Identifier for the job.
+
+`forecast_id`::
+  (string) A comma-separated list of forecast identifiers. 
+//TBD: Can this ID be retrieved somehow?
```
Comment (benwtrent) [inline docs/reference/ml/apis/delete-forecast.asciidoc]: ID is only available from the initial creation of the forecast. 

This API is most likely be utilized from the UI (and they know how to gather and display forecasts from the `.ml-shared` index, or as an "oops, I created a forecast via API with the wrong settings".

File: docs/reference/ml/apis/delete-forecast.asciidoc [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  If you do not specify this optional parameter or if you specify `_all`, the 
+  API deletes all forecasts from the job.  
+
+==== Request Parameters
+
+`allow_no_forecasts`::
+  (boolean) Whether to ignore if `_all` matches no forecasts. 
+//TBD: This description is unclear.  Does this mean that if this parameter is 
+//false and there are no forecasts in the job, an error is returned?
```
Comment (benwtrent) [inline docs/reference/ml/apis/delete-forecast.asciidoc]: Correct, if `_all` finds that there are NO forecasts, then an error is returned when this is `false`. Rare use case for sure, but is consistent with other APIs that allow this type of interaction.

File: docs/reference/ml/apis/delete-forecast.asciidoc [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+`timeout`::
+  (time units) Controls the time to wait until the forecasts are deleted. 
+  Defaults to 30 seconds. For more information about time units, see <<time-units>>.
+//TBD: This description is unclear.  Is it the time to wait until the deletion
```
Comment (benwtrent) [inline docs/reference/ml/apis/delete-forecast.asciidoc]: This is to wait for the deletion to be complete.

File: docs/reference/ml/apis/delete-forecast.asciidoc (hunk 1 above)
Comment (lcawl) [reply docs/reference/ml/apis/delete-forecast.asciidoc]: OK, so there are no immediate plans for a "GET forecasts" API?

Comment (benwtrent) [reply docs/reference/ml/apis/delete-forecast.asciidoc]: @lcawl no, there is not.

**Review by benwtrent — APPROVED** [review]
👍 Thanks a bunch for writing this up @lcawl !!!

**Pull request merged**